usage/src/org/apache/cloudstack/quota/QuotaManagerImpl.java

```diff
@@ -0,0 +1,465 @@
+                            quotalistforaccount.add(updateQuotaAllocatedVMUsage(usageRecord, aggregationRatio));
+                            break;
+                        case QuotaTypes.SNAPSHOT:
+                            quotalistforaccount.add(updateQuotaDiskUsage(usageRecord, aggregationRatio, QuotaTypes.SNAPSHOT));
+                            break;
+                        case QuotaTypes.TEMPLATE:
+                            quotalistforaccount.add(updateQuotaDiskUsage(usageRecord, aggregationRatio, QuotaTypes.TEMPLATE));
+                            break;
+                        case QuotaTypes.ISO:
+                            quotalistforaccount.add(updateQuotaDiskUsage(usageRecord, aggregationRatio, QuotaTypes.ISO));
+                            break;
+                        case QuotaTypes.VOLUME:
+                            quotalistforaccount.add(updateQuotaDiskUsage(usageRecord, aggregationRatio, QuotaTypes.VOLUME));
+                            break;
+                        case QuotaTypes.VM_SNAPSHOT:
+                            quotalistforaccount.add(updateQuotaDiskUsage(usageRecord, aggregationRatio, QuotaTypes.VM_SNAPSHOT));
+                            break;
+                        case QuotaTypes.LOAD_BALANCER_POLICY:
+                            quotalistforaccount.add(updateQuotaRaw(usageRecord, aggregationRatio, QuotaTypes.LOAD_BALANCER_POLICY));
+                            break;
+                        case QuotaTypes.PORT_FORWARDING_RULE:
+                            quotalistforaccount.add(updateQuotaRaw(usageRecord, aggregationRatio, QuotaTypes.PORT_FORWARDING_RULE));
+                            break;
+                        case QuotaTypes.IP_ADDRESS:
+                            quotalistforaccount.add(updateQuotaRaw(usageRecord, aggregationRatio, QuotaTypes.IP_ADDRESS));
+                            break;
+                        case QuotaTypes.NETWORK_OFFERING:
+                            quotalistforaccount.add(updateQuotaRaw(usageRecord, aggregationRatio, QuotaTypes.NETWORK_OFFERING));
+                            break;
+                        case QuotaTypes.SECURITY_GROUP:
+                            quotalistforaccount.add(updateQuotaRaw(usageRecord, aggregationRatio, QuotaTypes.SECURITY_GROUP));
+                            break;
+                        case QuotaTypes.VPN_USERS:
+                            quotalistforaccount.add(updateQuotaRaw(usageRecord, aggregationRatio, QuotaTypes.VPN_USERS));
+                            break;
+                        case QuotaTypes.NETWORK_BYTES_RECEIVED:
+                            quotalistforaccount.add(updateQuotaNetwork(usageRecord, QuotaTypes.NETWORK_BYTES_RECEIVED));
+                            break;
+                        case QuotaTypes.NETWORK_BYTES_SENT:
+                            quotalistforaccount.add(updateQuotaNetwork(usageRecord, QuotaTypes.NETWORK_BYTES_SENT));
```

Comment:
Could the ``QuotaTypes.SNAPSHOT```, ``QuotaTypes.TEMPLATE``, ``QuotaTypes.ISO``, ``QuotaTypes.VOLUME``, ``QuotaTypes.VM_SNAPSHOT``, ``QuotaTypes.LOAD_BALANCER_POLICY``, ``QuotaTypes.PORT_FORWARDING_RULE``, ``QuotaTypes.IP_ADDRESS``, ``QuotaTypes.NETWORK_OFFERING``,  ``QuotaTypes.SECURITY_GROUP``, ``QuotaTypes.VPN_USERS``, ``QuotaTypes.NETWORK_BYTES_RECEIVED``, ``QuotaTypes.NETWORK_BYTES_SENT`` cases be consolidated into the following:

```
quotalistforaccount.add(updateQuotaRaw(usageRecord, aggregationRatio, usageRecord.getUsageType()));
break;
```